rest/model/patch.go

```diff
@@ -150,13 +153,24 @@ func (apiPatch *APIPatch) BuildFromService(p patch.Patch) error {
 	}
 
 	projectIdentifier := p.Project
-	if p.Project != "" {
+	if withArgs && p.Project != "" {
 		identifier, err := model.GetIdentifierForProject(p.Project)
 		if err != nil {
 			return errors.Wrapf(err, "getting project '%s'", p.Project)
 		}
 		apiPatch.ProjectIdentifier = utility.ToStringPtr(identifier)
 		projectIdentifier = identifier
+
+		if p.IsCommitQueuePatch() {
+			cq, err := commitqueue.FindOneId(p.Project)
+			if err != nil {
+				return errors.Wrap(err, "error getting commit queue position")
+			}
+			apiPatch.CommitQueuePosition = utility.ToIntPtr(-1)
+			if cq != nil {
+				apiPatch.CommitQueuePosition = utility.ToIntPtr(cq.FindItem(p.Id.Hex()))
+			}
+		}
```

Comment:
It seems to me that all this is doing is setting another field on the patch. Can you please explain more about where this risk that you mentioned in the description is? 

> verify that not always getting project identifier isn't going to cause issues, in which case I can refactor

Reply:
This isn't referencing CommitQueuePosition, it's referencing identifier. As a part of this ticket I'm changing the behavior so if withArgs isn't specified, we just translate Patch -> APIPatch, and if withArgs is specified, then we get things from the DB to add to APIPatch (we're using this format slightly more for BuildFromService things).

Right now I'm setting withArgs to true for rest route functions only, i.e. for resolvers identifier won't be set. I'd like to verify that this isn't an issue for any existing UI patch usage, or if I should refactor this to consider identifier / commit queue position separately.

Reply:
I understand better now, thank you!